Goal: Validate player stats data by comparing with 'actual' data obtained from credible sources
Credible sources include: Matchroom Sports aggregate data and aggregate data from 'Atlarge', a user on AZbilliards forum

(unfinished)

In [8]:
import pandas as pd

In [5]:
import pickle

In [3]:
df= pd.read_pickle(path)

In [15]:
# Check data on 2016 American players
usa = ['Shane Van Boening', 'Rodney Morris', 'Mike Dechaine', 'Justin Bergman', 'Skyler Woodward']

['Shane Van Boening', 'am', 25, 10, 15, '10-15', 40.0, 25, 11, 14, '11-14', 44.0, '4-6', '0-0', 60, 25, 35, '25-35', 41.67]
['Rodney Morris', 'am', 18, 10, 8, '10-8', 55.56, 29, 12, 17, '12-17', 41.38, '3-5', '2-0', 57, 27, 30, '27-30', 47.37]
['Mike Dechaine', 'am', 7, 1, 6, '1-6', 14.29, 10, 2, 8, '2-8', 20.0, '2-2', '0-0', 21, 5, 16, '5-16', 23.81]
['Justin Bergman', 'am', 6, 1, 5, '1-5', 16.67, 7, 4, 3, '4-3', 57.14, '1-2', '0-0', 16, 6, 10, '6-10', 37.5]
['Skyler Woodward', 'am', 3, 0, 3, '0-3', 0.0, 6, 3, 3, '3-3', 50.0, '1-1', '0-0', 11, 4, 7, '4-7', 36.36]


### Data obtained from atlarge post on AZ billiards forum

Team USA
• Shane Van Boening 10-15 singles, 11-14 doubles, 4-6 teams, 25-35 total (42%)

• Rodney Morris 10-8 singles, 12-17 doubles, 2-0 trebles, 3-5 teams, 27-30 total (47%)

• Mike Dechaine 1-6 singles, 2-8 doubles, 2-2 teams, 5-16 total (24%)

• Justin Bergman  1-5 singles, 4-3 doubles, 1-2 teams, 6-10 total (38%)

• Skyler Woodward  0-3 singles, 3-3 doubles, 1-1 teams, 4-7 total (36%)

In [16]:
# Check stats for 2016 Eu players
eu = ['Darren Appleton', 'Mark Gray', 'Albin Ouschan', 'Niels Feijen']

['Darren Appleton', 'u', 14, 10, 4, '10-4', 71.43, 25, 10, 15, '10-15', 40.0, '5-3', '0-0', 47, 25, 22, '25-22', 53.19]
['Mark Gray', 'u', 2, 1, 1, '1-1', 50.0, 9, 4, 5, '4-5', 44.44, '3-0', '0-0', 14, 8, 6, '8-6', 57.14]
['Albin Ouschan', 'u', 3, 3, 0, '3-0', 100.0, 6, 5, 1, '5-1', 83.33, '1-1', '0-0', 11, 9, 2, '9-2', 81.82]
['Niels Feijen', 'u', 20, 12, 8, '12-8', 60.0, 33, 24, 9, '24-9', 72.73, '5-4', '0-0', 62, 41, 21, '41-21', 66.13]


### Data obtained from atlarge post on AZ billiards forum

Team Europe
• Niels Feijen 12-8 singles, 24-9 doubles, 5-4 teams, 41-21 total (66%)

• Darren Appleton  10-4 singles, 10-15 doubles, 5-3 teams, 25-22 total (53%)

• Mark Gray   1-1 singles, 4-5 doubles, 3-0 teams, 8-6 total (57%)

• Albin Ouschan  3-0 singles, 5-1 doubles, 1-1 teams, 9-2 total (82%)

• Jayson Shaw 1-1 singles, 1-1 doubles, 1-0 teams, 3-2 total (60%)

Data obtained from atlarge post on AZ billiards forum

### More info to verify:
• Johnny Archer (17 years) ..... 18-13 singles, 26-23 doubles, 0-2 trebles, 5-4 teams, 49-42 total (winning percentage of 54%)
• Earl Strickland (14 years) ..... 14-12 singles, 24-12 doubles, 1-1 trebles, 3-2 teams, 42-27 total (61%)
• Corey Deuel (9 years) ..... 7-6 singles, 13-12 doubles, 2-0 trebles, 5-2 teams, 27-20 total (57%)
And more:
Dennis Hatch (4 years -- 2009, 2010, 2012, 2013) ..... 4-3 singles, 5-7 doubles, 2-2 teams, 11-12 total (winning percentage 48%)

In [17]:
pl=['Johnny Archer','Earl Strickland','Corey Deuel','Dennis Hatch']
for player in pl:
    print(winlosslist(player,'am'))

['Johnny Archer', 'am', 30, 17, 13, '17-13', 56.67, 50, 27, 23, '27-23', 54.0, '5-4', '1-1', 91, 50, 41, '50-41', 54.95]
['Earl Strickland', 'am', 26, 14, 12, '14-12', 53.85, 36, 24, 12, '24-12', 66.67, '3-2', '2-0', 69, 43, 26, '43-26', 62.32]
['Corey Deuel', 'am', 13, 7, 6, '7-6', 53.85, 25, 13, 12, '13-12', 52.0, '5-2', '2-0', 47, 27, 20, '27-20', 57.45]
['Dennis Hatch', 'am', 7, 4, 3, '4-3', 57.14, 12, 5, 7, '5-7', 41.67, '2-2', '0-0', 23, 11, 12, '11-12', 47.83]


# note: archer and strickland are off by one (too many wins) 
The source on AZ billiards has got the wrong triples info. Only triples matches occured in 2006 and johnny was 1-1 earl was 2-0 (source wikipedia)

### Data obtained from atlarge post on AZ billiards forum

Data including 2016

Team USA
• Shane Van Boening (10 years -- 2007-2016) ..... 10-15 singles, 11-14 doubles, 4-6 teams, 25-35 total (winning percentage of 42%)

• Rodney Morris (10 years -- 2003-2008, 2010-2011, 2013, 2016) ..... 10-8 singles, 12-17 doubles, 2-0 trebles, 3-5 teams, 27-30 total (47%)

• Mike Dechaine (4 years -- 2011-2012, 2015-2016) ..... 1-6 singles, 2-8 doubles, 2-2 teams, 5-16 total (24%)

• Justin Bergman (3 years -- 2014-2016) ..... 1-5 singles, 4-3 doubles, 1-2 teams, 6-10 total (38%)

• Skyler Woodward (2 years -- 2015-2016) ..... 0-3 singles, 3-3 doubles, 1-1 teams, 4-7 total (36%)

• Team USA Total ..... 22-37 singles, 32-45 doubles, 2-0 trebles, 11-16 teams, 67-98 total (41%)

Team Europe
• Niels Feijen (12 years -- 2001, 2004-2005, 2007-2009, 2011-2016) ..... 12-8 singles, 24-9 doubles, 5-4 teams, 41-21 total (66%)

• Darren Appleton (8 years -- 2009-2016) ..... 10-4 singles, 10-15 doubles, 5-3 teams, 25-22 total (53%)

• Mark Gray (3 years -- 2008, 2014, 2016) ..... 1-1 singles, 4-5 doubles, 3-0 teams, 8-6 total (57%)

• Albin Ouschan (2 years -- 2015-2016) ..... 3-0 singles, 5-1 doubles, 1-1 teams, 9-2 total (82%)

• Jayson Shaw (1 year -- 2016) ..... 1-1 singles, 1-1 doubles, 1-0 teams, 3-2 total (60%)

• Team Europe Total ..... 27-14 singles, 44-31 doubles, 15-8 teams, 86-53 total (62%)